helm/distributed-idol/templates/content/statefulset.yaml

```diff
@@ -1,109 +1,49 @@
-        volumeMounts:
-        - name: config-map
-          mountPath: /etc/config/idol
-        - name: index
-          mountPath: /opt/idol/content/index
-{{- if .Values.setupMirrored }}
-        - name: archive-share
-          mountPath: /opt/idol/archive
-{{- end }}
-        {{- range $component.additionalVolumeMounts }}
-        - {{ . | toYaml | nindent 10 }}
-        {{- end }}
-        env:
-        - name: IDOL_CONTENT_SERVICE_PORT_ACI_PORT
-          # This doesn't seem to be set by our headless service
-          value: {{ $component.aciPort | quote }}
-        - name: IDOL_COMMAND_PARAMS
-          value: "-idolcluster"
-        {{- if $component.envConfigMap }}
-        envFrom:
-        - configMapRef: {{ $component.envConfigMap | quote }}
-        {{ end }}
-        lifecycle:
-          postStart:
-            exec:
-              command:
-              - /bin/bash
-              - "-c"
-              - |
-{{ tpl (.Files.Get "resources/content_postStart.sh") . | indent 16 }}
-          preStop:
-            exec:
-              command:
-              - /bin/bash
-              - "-c"
-              - |
-{{ tpl (.Files.Get "resources/content_preStop.sh") . | indent 16 }}
-        livenessProbe:
-          httpGet:
-            path: /a=getpid
-            port: {{ $component.aciPort | int }}
-{{- template "idol-library.standardLivenessProbe" $component.livenessProbe }}
-        ports:
-        - containerPort: {{ $component.aciPort | int }}
-          name: aci-port
-          protocol: TCP
-        - containerPort: {{ $component.servicePort | int }}
-          name: service-port
-          protocol: TCP
-        - containerPort: {{ $component.indexPort | int }}
-          name: index-port
-          protocol: TCP
-      volumes:
-      - name: config-map
-        configMap:
-          name: {{ default (printf "%s-default-cfg" $component.name) $component.existingConfigMap }}
-{{- if .Values.setupMirrored }}
-      - name: archive-share
-        persistentVolumeClaim:
-          claimName: backup-archive-pvc
-{{- end }}
-      {{- range $component.additionalVolumes }}
-      - {{ . | toYaml | nindent 8 }}
-      {{- end }}
   volumeClaimTemplates:
   - metadata:
       name: index
-      labels: {{- include "distributedidol.labels" . | nindent 8 }}
+      labels: {{- include "distributedidol.labels" $root | nindent 8 }}
     spec:
       accessModes: [ "ReadWriteOnce" ]
       storageClassName: {{ $component.contentStorageClass }}
       resources:
         requests:
-          storage: {{ $component.contentVolumeSize }}
\ No newline at end of file
+          storage: {{ $component.contentVolumeSize }}
+{{- end }}
+
+{{- $contentVolumes := list }}
+{{- if .Values.setupMirrored }}
+{{- $contentVolumes = append $contentVolumes (dict
+   "name" "archive-share"
+   "persistentVolumeClaim" (dict "claimName" "backup-archive-pvc")
+) }}
+{{- end }}
+
+{{- $contentValues := merge .Values.content (dict "global" .Values.global) }}
```

Comment:
You shouldn't need to do this. Helm global values are really global and you shouldn't need to move them around.

Reply:
In that case I guess I should change [this idol-library template](https://github.com/opentext-idol/idol-containers-toolkit/blob/main/helm/idol-library/templates/aciserver/_container.tpl#L24) to be using `$root.Values.global.imagePullPolicy` instead of `$component.global.imagePullPolicy`? 

Without the above change I get `nil pointer evaluating interface {}.imagePullPolicy`.